src/main.cpp

```diff
@@ -185,16 +161,37 @@ main(int argc, char *argv[])
         if (parser.isSet(debugOption))
                 nhlog::enable_debug_log_from_commandline = true;
 
+        QString appNameWithProfile{"nheko"};
+        if (parser.isSet(configName))
+                appNameWithProfile.append(QString{"-%1"}.arg(parser.value(configName)));
+
+        QSharedMemory mem{appNameWithProfile};
```

Comment:
Do we still need SingleApplication then?

Reply:
Probably not.